agent/engine/docker_task_engine_test.go

```diff
@@ -3345,7 +3346,7 @@ func TestPeriodicExecAgentsMonitoring(t *testing.T) {
 			defer topCancel()
 			return resp, nil
 		}).AnyTimes()
-	go taskEngine.(*DockerTaskEngine).startPeriodicExecAgentsMonitoring(ctx, 2*time.Millisecond)
+	go taskEngine.(*DockerTaskEngine).startPeriodicExecAgentsMonitoring(ctx)
```

Comment:
just curious, why remove `duration` as an argument? taking it as an input is seemingly more flexible

Reply:
Good question. I change this because `startPeriodicExecAgentsMonitoring` is invoked in the engine's `Init` method.
So the parameter might be more flexible to test `startPeriodicExecAgentsMonitoring` in isolation, but it makes it difficult to test the engine as a whole, since there's little control as to when this private method is going to be invoked. 
With the timeout as an attribute of the engine, it doesn't really matter when `startPeriodicExecAgentsMonitoring` is invoked internally.

Reply:
thanks for the explanation!